### DISCUSSION TITLE
Ability to connect to MQTT broker with password / username lock

### DISCUSSION THREAD

**Discussion by AHorvat-GIS** [brugr9/UEPluginIntegrationTool#6] [Q&A] [answered]
Dear Mr. Bruggmann,

We are currently using this tool here at Gallarus Industry Solutions for development of digital twinning and we are using this plugin.
I'm just wondering would it be possible to connect to an MQTT broker that has a password / username lock? The pub / sub socket's endpoint properties only display protocol, host and port fields.

Please get back to me as soon as possible on my e-mail ahorvat@intern.gis.ie

Kind Regards
Alex Horvat

**Comment by brugr9** [comment]
Dear Mr. Horvat,

Thanks for your interest in the plugin. 

Well, please note that the NNG&trade; software library does not support the MQTT transport protocol mqtt:// yet, so neither do the NNG sockets in the plugin.

If the broker accepts connections over tcp://, then it depends on the broker which payload should be sent (e.g., JSON or XML, structure of the fields etc.). But this is no longer the actual matter of the plugin. In addition please remark that using tcp:// the password is sent unencrypted.

Kindest regards,
@brugr9

**Reply by AHorvat-GIS** [reply]
Dear Mr Bruggmann,

Thank you for your swift response,
On the note of NNG not supporting MQTT transport protocol, I am wondering if this plugin can connect to an EMQX cloud or local EMQX broker. I understand this is outside the scope of this discussion and i'm not familiar with NNG as a whole but i'd appreciate if you had an answer to this question. 

Kind Regards
Alex Horvat

**Comment by brugr9** [comment, marked as answer]
Dear Mr. Horvat,

I have checked with EMQ (thanks again to @JaylinYu ). 
According to the manufacturer, one can use MQTT messaging broker @NanoMQ as a proxy:

* See _**"NanoMQ nng message proxy"**_, In: [https://github.com/emqx/nanomq#quick-start](https://github.com/emqx/nanomq#quick-start)
* For, e.g., a Docker image, visit: [https://www.emqx.com/en/try?product=nanomq](https://www.emqx.com/en/try?product=nanomq)

Kind Regards,
Roland Bruggmann

**Reply by AHorvat-GIS** [reply]
Dear Mr Bruggmann,
That is perfect, will develop this using nanomq.
Thank you for your brilliant support.

Kind Regards
Alex Horvat

**Comment by brugr9** [comment]
Dear Mr. Horvat,

As of version 5.1, UE provides with a built-in IOT plugin "MQTT". Note: The plugin is Beta (buggy) and is not yet documented, 
cp. brugr9/Heartbeat51#1

Kind Regards,
Roland Bruggmann

![ScreenshotPlugin](https://user-images.githubusercontent.com/1733987/214003095-2186743d-dc41-4b21-b045-450e33dfdccd.png)
![ProjectSettings_-_PluginMQTT](https://user-images.githubusercontent.com/1733987/214003091-fa744e4b-f457-4117-8d63-9e4f72a1ff98.png)
